### PR TITLE
Volcano speed improvements and Extreme Hills improvements

### PR DESCRIPTION
**Pull request by Zeno410**
The Volcano improvements are STRONGLY recommended. They make a HUGE difference.

### CODE REVIEW

**Comment by srs-bsns** [comment]
@Zeno410 ..should have fetched and rebased to resolve those conflicts before PRing. ;)

The volcano patch should be backported to 1.10, and soonish, so that there will be an updated 1.10 release for the next MMD dev server pack.

File: src/main/java/rtg/world/gen/VolcanoGenerator.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ *
+ * @author Zeno410
+ */
+public class VolcanoGenerator {
+    
+    private Random mapRand;
+    private long seed;
+    protected RTGConfig rtgConfig = RTGAPI.config();
+    private static long l;
+    private static long l1;
```
Comment (srs-bsns) [inline src/main/java/rtg/world/gen/VolcanoGenerator.java:29]: Could these not have been named better?

File: src/main/java/rtg/world/gen/ChunkProviderRTG.java [hunk 2]
```diff
@@ -311,7 +284,8 @@ public Chunk provideChunk(final int cx, final int cz) {
                 baseBiomesList[k] = biomePatcher.getPatchedBaseBiome("" + Biome.getIdForBiome(landscape.biome[k].baseBiome));
             }
         }
-
+        volcanoGenerator.generateMapGen(primer, worldSeed, worldObj, cmr, mapRand, cx, cz, rtgWorld.simplex, rtgWorld.cell, landscape.noise);
```
Comment (srs-bsns) [inline src/main/java/rtg/world/gen/ChunkProviderRTG.java:287]: This is the point where we should be checking if volcanos are enabled in the configs.
If enabled, run the generator, if not, kick rocks.

**Comment by Zeno410** [comment]
I DID rebase and I still got the conflicts.

**Comment by Zeno410** [comment]
The generator checks before it does anything significant, so it's not a performance issue.

**Pull request closed without merging**